src/west/manifest-schema.yml

```diff
@@ -95,6 +95,21 @@ mapping:
           path:
             required: false
             type: str
+          # Submodules to checkout within the project
+          submodules:
+            required: false
+            type: seq
+            sequence:
+              - type: map
+                mapping:
+                  # Submodule's unique name.
+                  name:
```

Comment:
Is the name used anywhere? Or requested by Carles? :)

Reply:
Basically it is not used to any key functional actions, as path. Anyway I think that it is nice to have name for some object, to be able to find it easier or log some errors related to this. Don't you think it would look weird to have list of paths without names, especially when paths may not always look so obviously? For example in NCS there are submodules unity and cmock, which are located on the throwtheswitch remote and unity is checkout inside of the cmock, so for cmock path is test/cmock and for unity it is tests/cmock/vendor/unity :smile:

Reply:
I don't know. On one hand I get your point, but on the other hand I would prefer having full submodule paths in logs etc. so I don't have to do the mapping myself whenever a problem occurs. Which makes me think that if we don't have a use case for more complex structure than maybe it's better to simplify it. But let the west community decide ;)